batch client panic when close

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```log
```
RunBrBackup: cmd start at 2024-05-30 14:49:31
cmd failed at 2024-05-30 14:58:32
stdout:
Detail BR log in /tmp/br.log.2024-05-30T06.49.31Z 
panic: close of closed channel

goroutine 2822 [running]:
[github.com/tikv/client-go/v2/internal/client.(*batchConn).Close(...)](http://github.com/tikv/client-go/v2/internal/client.(*batchConn).Close(...))
        /root/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client_batch.go:769](http://github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client_batch.go:769)
[github.com/tikv/client-go/v2/internal/client.(*connArray).Close(0xc0004bb560)](http://github.com/tikv/client-go/v2/internal/client.(*connArray).Close(0xc0004bb560))
        /root/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client.go:330](http://github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client.go:330) +0xf6
[github.com/tikv/client-go/v2/internal/client.(*RPCClient).CloseAddr](http://github.com/tikv/client-go/v2/internal/client.(*RPCClient).CloseAddr)(0xc0007e3680, {0xc000f7e7d0, 0x44})
        /root/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client.go:797](http://github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client.go:797) +0x1e5
[github.com/tikv/client-go/v2/internal/client.(*RPCClient).recycleIdleConnArray(0xc0007e3680)](http://github.com/tikv/client-go/v2/internal/client.(*RPCClient).recycleIdleConnArray(0xc0007e3680))
        /root/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client_batch.go:835](http://github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client_batch.go:835) +0x1f7
created by [github.com/tikv/client-go/v2/internal/client.(*RPCClient).sendRequest](http://github.com/tikv/client-go/v2/internal/client.(*RPCClient).sendRequest) in goroutine 2821
        /root/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client.go:587](http://github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706/internal/client/client.go:587) +0x34b
```
```

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

release-7.5: 84e292668495cba54fc5a7a157b7e0519c4267e4

--- a/DEPS.bzl
+++ b/DEPS.bzl
@@ -7041,13 +7041,13 @@ def go_deps():
         name = "com_github_tikv_client_go_v2",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/tikv/client-go/v2",
-        sha256 = "02e51a0a3f3737b365c2a16e7147c472b99ad1c48a285ba840b55a59d20cf729",
-        strip_prefix = "github.com/tikv/client-go/v2@v2.0.8-0.20240528025037-f41b068bf706",
+        sha256 = "48097286e81b799d8cc8612879e4d3fa569be79a1be93ed77131c04052c3a069",
+        strip_prefix = "github.com/tikv/client-go/v2@v2.0.8-0.20240531122021-7a74511a5241",
         urls = [
-            "http://bazel-cache.pingcap.net:8080/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240528025037-f41b068bf706.zip",
-            "http://ats.apps.svc/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240528025037-f41b068bf706.zip",
-            "https://cache.hawkingrei.com/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240528025037-f41b068bf706.zip",
-            "https://storage.googleapis.com/pingcapmirror/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240528025037-f41b068bf706.zip",
+            "http://bazel-cache.pingcap.net:8080/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240531122021-7a74511a5241.zip",
+            "http://ats.apps.svc/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240531122021-7a74511a5241.zip",
+            "https://cache.hawkingrei.com/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240531122021-7a74511a5241.zip",
+            "https://storage.googleapis.com/pingcapmirror/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240531122021-7a74511a5241.zip",
         ],
     )
     go_repository(

--- a/go.mod
+++ b/go.mod
@@ -104,7 +104,7 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/tdakkota/asciicheck v0.2.0
 	github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2
-	github.com/tikv/client-go/v2 v2.0.8-0.20240528025037-f41b068bf706
+	github.com/tikv/client-go/v2 v2.0.8-0.20240531122021-7a74511a5241
 	github.com/tikv/pd/client v0.0.0-20240528122050-634e05a87ee0
 	github.com/timakin/bodyclose v0.0.0-20230421092635-574207250966
 	github.com/twmb/murmur3 v1.1.6

